python/ray/serve/_private/client.py

```diff
@@ -107,27 +106,21 @@ def shutdown(self) -> None:
             del self.handle_cache[k]
 
         if ray.is_initialized() and not self._shutdown:
-            ray.get(self._controller.shutdown.remote())
-            self._wait_for_deployments_shutdown()
-
-            ray.kill(self._controller, no_restart=True)
-
-            # Wait for the named actor entry gets removed as well.
-            started = time.time()
-            while True:
-                try:
-                    ray.get_actor(self._controller_name, namespace=SERVE_NAMESPACE)
-                    if time.time() - started > 5:
-                        logger.warning(
-                            "Waited 5s for Serve to shutdown gracefully but "
-                            "the controller is still not cleaned up. "
-                            "You can ignore this warning if you are shutting "
-                            "down the Ray cluster."
-                        )
-                        break
-                except ValueError:  # actor name is removed
-                    break
+            try:
+                controller_shutdown_ref = (
+                    self._controller.await_controller_shutdown.remote()
+                )
+                ray.get(self._controller.set_shutting_down_flag.remote())
+                logger.info(
+                    "Serve controller received shutdown request. It "
+                    "will shut down gracefully in the background."
+                )
 
+                self._wait_for_deployments_shutdown()
+                # Wait until the named actor entry gets removed as well.
+                ray.get(controller_shutdown_ref)
```

Comment:
This looks unnecessarily complicated, why isn't as simple as this?
```suggestion
                ray.get(self._controller.shutdown.remote(), timeout_s=10)
```

Reply:
Mostly bc I don't know how `asyncio.Event()` works XD Thanks for the suggestion!